From f1d32420d477601ea34562b052c6863bb3f055b0 Mon Sep 17 00:00:00 2001
From: Dillibabu <dillibabug@ami.com>
Date: Mon, 23 Sep 2024 16:41:28 +0530
Subject: [PATCH] 1. Added new property for CertificateType will hold
 certifiate type. 2. Added new property for CertificateString will hold all
 certificate bytes

Signed-off-by: Dillibabu <dillibabug@ami.com>
---
 .../Certs/Certificate.interface.yaml          | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/yaml/xyz/openbmc_project/Certs/Certificate.interface.yaml b/yaml/xyz/openbmc_project/Certs/Certificate.interface.yaml
index a19bb44..c993f03 100644
--- a/yaml/xyz/openbmc_project/Certs/Certificate.interface.yaml
+++ b/yaml/xyz/openbmc_project/Certs/Certificate.interface.yaml
@@ -6,6 +6,15 @@ description: >
     xyz.openbmc_project.Object.Delete to allow the deletion of individual
     certificate objects.
 properties:
+    - name: CertificateType
+      type: enum[self.Type]
+      description: >
+           Enum will hold the certificate type, type could be a
+           Single pem, Chain pem or PKCS type.
+    - name: ChainCertString
+      type: string
+      description: >
+           String will hold the chain certificate data.
     - name: CertificateString
       type: string
       description: >
@@ -119,3 +128,18 @@ associations:
       reverse_name: responder_identified_by
       required_endpoint_interfaces:
           - xyz.openbmc_project.Attestation.IdentityAuthentication
+
+enumerations:
+    - name: Type
+      description: >
+          Certificate types.
+      values:
+          - name: PEM
+            description: >
+                Single pem certificate.
+          - name: PEMchain
+            description: >
+                PEM Chain certificate.
+          - name: PKC11
+            description: >
+                PKC11 implemetation not available.
-- 
2.34.1

